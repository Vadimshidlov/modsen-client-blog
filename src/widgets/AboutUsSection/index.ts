export { AboutUsSection } from '@/widgets/AboutUsSection/ui/AboutUsSection';
