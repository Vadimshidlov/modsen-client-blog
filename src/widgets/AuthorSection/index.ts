export { AuthorSection } from '@/widgets/AuthorSection/ui/AuthorSection';
