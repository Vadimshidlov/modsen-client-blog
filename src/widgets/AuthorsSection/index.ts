export { AuthorsSection } from '@/widgets/AuthorsSection/ui/AuthorsSection';
