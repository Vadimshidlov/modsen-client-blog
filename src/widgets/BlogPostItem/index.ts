export { BlogPostItem } from '@/widgets/BlogPostItem/BlogPostItem';
