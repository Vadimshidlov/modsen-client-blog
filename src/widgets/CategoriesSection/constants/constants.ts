export const SEARCH_TAGS = ['Business', 'Experience', 'Screen', 'Technology', 'Marketing', 'Life'];
