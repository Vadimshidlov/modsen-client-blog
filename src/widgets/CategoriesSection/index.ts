export { CategoriesSection } from '@/widgets/CategoriesSection/ui/CategoriesSection';
