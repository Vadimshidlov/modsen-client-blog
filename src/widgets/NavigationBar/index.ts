export { NavigationBar } from '@/widgets/NavigationBar/ui/NavigationBar';
