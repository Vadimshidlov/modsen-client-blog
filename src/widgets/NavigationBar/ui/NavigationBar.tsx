import React from 'react';
import Link from 'next/link';
import { useTranslations } from 'next-intl';
import { NavigationBarPropsType } from '@/shared/types';
import styles from '@/widgets/NavigationBar/ui/NavigationBar.module.scss';

export function NavigationBar({ navigationLinks }: NavigationBarPropsType) {
  const t = useTranslations('NavigationBar');

  return (
    <ul className={styles.navigation}>
      {navigationLinks.map(({ text, link }) => (
        <li className={styles.navigation__item} key={text}>
          <Link key={text} href={link} data-testid={`${text.toLowerCase()}-header-link`}>
            {t(`${text}`)}
          </Link>
        </li>
      ))}
    </ul>
  );
}
