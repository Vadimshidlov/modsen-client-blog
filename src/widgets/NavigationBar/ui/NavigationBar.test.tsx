import React from 'react';
import { render } from '@testing-library/react';
import { NavigationBar } from '@/widgets/NavigationBar';
import { NAVIGATION_LINKS } from '@/shared/constants/navigationBar';

jest.mock('next-intl', () => ({
  useTranslations: jest.fn().mockReturnValue((key: string) => `Mocked ${key}`),
}));

describe('NavigationBar tests', () => {
  test('should matches snapshot', () => {
    const { container } = render(<NavigationBar navigationLinks={NAVIGATION_LINKS} />);
    expect(container).toMatchSnapshot();
  });
});
