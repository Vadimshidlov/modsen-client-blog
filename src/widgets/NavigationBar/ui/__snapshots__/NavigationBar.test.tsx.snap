// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar tests should matches snapshot 1`] = `
<div>
  <ul
    class="navigation"
  >
    <li
      class="navigation__item"
    >
      <a
        data-testid="home-header-link"
        href="/"
      >
        Mocked Home
      </a>
    </li>
    <li
      class="navigation__item"
    >
      <a
        data-testid="blog-header-link"
        href="/blog"
      >
        Mocked Blog
      </a>
    </li>
    <li
      class="navigation__item"
    >
      <a
        data-testid="about us-header-link"
        href="/about"
      >
        Mocked About Us
      </a>
    </li>
    <li
      class="navigation__item"
    >
      <a
        data-testid="contact us-header-link"
        href="/"
      >
        Mocked Contact us
      </a>
    </li>
    <li
      class="navigation__item"
    >
      <a
        data-testid="privacy policy-header-link"
        href="/privacy"
      >
        Mocked Privacy Policy
      </a>
    </li>
  </ul>
</div>
`;
